laboratory-servlet-api-integration/src/it/java/com/wixpress/petri/test/LaboratoryIT.java

```diff
@@ -141,4 +140,13 @@ public void reportsSingleSpecToPetri() throws Exception {
         assertThat(specs.get(0).getScopes(), CoreMatchers.is(Arrays.asList(ValidStubSpecDefinition_1.scopeDefinitions)));
     }
 
+    @Test
+    public void resilientToPetriServerDowntime() throws Exception {
```

Comment:
I get your point.
test name should then maybe be 'cahcesExperiments' or 'doesNotFetchExperimentOnUserThread' or similar?

the one thing that bothers me here is that the test failed and did drive the change, but now if you read it or even comment out the 'failNextRequest' it will still be green.
is that cool 'TDD wise'?